src/panelMenuButton.js

```diff
@@ -116,10 +123,13 @@ var PanelMenuButton = GObject.registerClass(
                         }
 
                         if (!this.isRunnerHidden) {
-                            const isRunningSpriteShown = this.cpu.utilization > this.sleepingThreshold;
-                            this.ui.get('icon').set_gicon(
-                                isRunningSpriteShown ? this.iconProvider.nextSprite : this.iconProvider.sleeping,
-                            );
+                            if (this.cpu.utilization > this.sleepingThreshold) {
+                                this.iconProvider.animationState = states.RUNNING;
+                            } else {
+                                this.iconProvider.animationState = this.animatedSleeping
+                                    ? states.SLEEPING : states.SLEEPING_STATIC;
+                            }
```

Comment:
`iconProvider` might need to handle this state management code in order to avoid using `'states'` outside its module.